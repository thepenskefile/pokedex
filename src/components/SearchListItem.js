import React from 'react';
// import { Link } from '@reach/router';
import { Box, Column, Columns, styled, palette } from 'fannypack';
import { Link } from '@reach/router';
import { CONTENT_CATEGORY_TYPES } from '../_types/content_category_types';

const StyledListItem = styled(Box)`
  border-bottom: 1px solid ${palette('white800')};
  padding-bottom: 20px;
  padding-top: 20px;
  padding-left: 10px;
  &:hover {
    background-color: ${palette('white700')};
  }
`;

type Props = {
  index: Number,
  name: string,
  id: Number,
  category: string
};

const SearchListItem = ({ index, name, id, category }: Props) => (
  <Link to={`/${CONTENT_CATEGORY_TYPES[category]}/${id}`}>
    <StyledListItem>
      <Columns minBreakpoint="mobile">
        <Column spread={2}>{index}</Column>
        <Column spread={10}>{name.charAt(0).toUpperCase() + name.slice(1)}</Column>
      </Columns>
    </StyledListItem>
  </Link>
);

export default SearchListItem;
