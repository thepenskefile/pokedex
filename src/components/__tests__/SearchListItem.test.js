import React from 'react';
import SearchListItem from '../SearchListItem';
import renderer from 'react-test-renderer';

const defaultProps = {
  index: 'mockIndex',
  name: 'mockName',
  id: 'mockId'
};

it('renders correctly', () => {
  const component = renderer.create(<SearchListItem {...defaultProps} />);
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
