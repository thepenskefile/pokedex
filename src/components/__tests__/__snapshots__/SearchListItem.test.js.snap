// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<a
  href="/undefined/mockId"
  onClick={[Function]}
>
  <div
    className="sc-bdVaJa inEjad Box-sc-6sssmg-0 inpIDs"
  >
    <div
      className="styled-otaduv-0 wkSBA Box-sc-6sssmg-0 inpIDs"
    >
      <div
        className="styled__Column-sc-1lmgd0f-0 esYmZI Box-sc-6sssmg-0 inpIDs"
      >
        mockIndex
      </div>
      <div
        className="styled__Column-sc-1lmgd0f-0 RTVYE Box-sc-6sssmg-0 inpIDs"
      >
        mockName
      </div>
    </div>
  </div>
</a>
`;
